### DISCUSSION TITLE
Editor hotkey vs. button behavior oversight

### DISCUSSION THREAD

**Discussion by plaguemask** [ppy/osu#24539] [Q&A]
I've noticed an inconsistency with the behavior of the editor's rotation/flip hotkeys compared to their respective buttons.

When pressing the buttons, flips and rotations are done around the center of the _selection_. Previously, hotkeys all functioned the same as buttons. This changed with https://github.com/ppy/osu/pull/16328, making `Ctrl + H` and `Ctrl + J` flip the selection around the center of the _playfield_, matching stable.

The oversight: the change did not include the rotation hotkeys. `Ctrl + ,` and `Ctrl + .` still act equivalent to the buttons, rotating around the center of the selection, counter to expectations. The same rationale would apply to making the change to these hotkeys as well.

**Comment by peppy** [comment]
This is intentional as those hotkeys are both hidden and will likely only be used by stable users.

**Reply by plaguemask** [reply]
Isn't that all the more reason to make the hotkeys match stable's behavior?

My problem is that `Ctrl + ,` and `Ctrl + .` *don't* match stable, when I feel they should.

**Reply by Walavouchey** [reply]
even the "hidden" rotation hotkeys? as mentioned in op those behave differently from stable

the question of whether the stable behaviour would be preferred by mappers has yet to be assessed though.

**Reply by peppy** [reply]
Hmm, that's weird. We made others match. Maybe these need to be updated (may not have had the origin implemented previously).
